### ISSUE TITLE
ZRange behavior does not match Redis when using a negative STOP argument

### ISSUE THREAD

**Issue by sinclairtarget** [brigade/mock_redis#131]
Using `redis-cli`:
```
127.0.0.1:6379> ZADD myzset 10 10
(integer) 1
127.0.0.1:6379> ZADD myzset 20 20
(integer) 1
127.0.0.1:6379> ZADD myzset 30 30
(integer) 1
127.0.0.1:6379> ZRANGE myzset 0 -4
(empty list or set)
```

Using [`redis-rb`](https://github.com/redis/redis-rb):
```
irb(main):001:0> redis = Redis.new
=> #<Redis client v3.3.3 for redis://127.0.0.1:6379/0>
irb(main):002:0> redis.zadd('myzset', 10, 10)
=> true
irb(main):003:0> redis.zadd('myzset', 20, 20)
=> true
irb(main):004:0> redis.zadd('myzset', 30, 30)
=> true
irb(main):005:0> redis.zrange('myzset', 0, -4)
=> []
```

Using `mock_redis`, version 0.17.2:
```
irb(main):001:0> redis = MockRedis.new
=> #<MockRedis:0x007fb63b8e94c0 @options={:scheme=>"redis", :host=>"127.0.0.1", :port=>6379, :path=>nil, :timeout=>5.0, :password=>nil, :db=>0, :time_class=>Time}, @db=#<MockRedis::PipelinedWrapper:0x007fb63b8e9150 @db=#<MockRedis::TransactionWrapper:0x007fb63b8e91a0 @db=#<MockRedis::ExpireWrapper:0x007fb63b8e91f0 @db=#<MockRedis::MultiDbWrapper:0x007fb63b8e9380 @db_index=0, @prototype_db=#<MockRedis::Database:0x007fb63b8e9330 @base=#<MockRedis:0x007fb63b8e94c0 ...>, @data={}, @expire_times=[]>, @databases={0=>#<MockRedis::Database:0x007fb63b8e9420 @base=#<MockRedis:0x007fb63b8e94c0 ...>, @data={}, @expire_times=[]>}>>, @transaction_futures=[], @in_multi=false, @multi_block_given=false>, @pipelined_futures=[], @in_pipeline=false>>
irb(main):002:0> redis.zadd('myzset', 10, 10)
=> true
irb(main):003:0> redis.zadd('myzset', 20, 20)
=> true
irb(main):004:0> redis.zadd('myzset', 30, 30)
=> true
irb(main):005:0> redis.zrange('myzset', 0, -4)
=> ["10"]
```
It looks like #128 introduced the problem.

**Comment by sinclairtarget** [comment]
Just checked—this definitely isn't an issue in version 0.17.1:
```
irb(main):001:0> $redis
=> #<MockRedis:0x007fda302407e0 @options={:scheme=>"redis", :host=>"", :port=>0, :path=>nil, :timeout=>5.0, :password=>nil, :db=>0, :time_class=>Time, :connect_timeout=>nil, :read_timeout=>nil, :write_timeout=>nil}, @db=#<MockRedis::PipelinedWrapper:0x007fda30240330 @db=#<MockRedis::TransactionWrapper:0x007fda30240380 @db=#<MockRedis::ExpireWrapper:0x007fda302403a8 @db=#<MockRedis::MultiDbWrapper:0x007fda30240588 @db_index=0, @prototype_db=#<MockRedis::Database:0x007fda30240538 @base=#<MockRedis:0x007fda302407e0 ...>, @data={}, @expire_times=[]>, @databases={0=>#<MockRedis::Database:0x007fda30240650 @base=#<MockRedis:0x007fda302407e0 ...>, @data={}, @expire_times=[]>}>>, @transaction_futures=[], @in_multi=false, @multi_block_given=false>, @pipelined_futures=[], @in_pipeline=false>>
irb(main):002:0> $redis.zadd('myzset', 10, 10)
=> true
irb(main):003:0> $redis.zadd('myzset', 20, 20)
=> true
irb(main):004:0> $redis.zadd('myzset', 30, 30)
=> true
irb(main):005:0> $redis.zrange('myzset', 0, -4)
=> []
```

**Comment by sds** [comment]
Fixed by #132. Thanks for the report!
